dd-sdk-android/src/main/kotlin/com/datadog/android/rum/tracking/NavigationViewTrackingStrategy.kt

```diff
@@ -98,6 +129,44 @@ class NavigationViewTrackingStrategy(
 
     // endregion
 
+    // region Internal
+
+    // endregion
+
+    internal class NavControllerFragmentLifecycleCallbacks(
+        private val navController: NavController,
+        argumentsProvider: (Fragment) -> Map<String, Any?>,
+        componentPredicate: ComponentPredicate<Fragment>
+    ) : AndroidXFragmentLifecycleCallbacks(
+        argumentsProvider,
+        componentPredicate,
+        rumMonitor = NoOpRumMonitor(),
+        advancedRumMonitor = AdvancedMonitorDecorator(
+            GlobalRum.get() as? AdvancedRumMonitor ?: NoOpAdvancedRumMonitor()
+        )
+    ) {
+        override fun resolveKey(fragment: Fragment): Any {
```

Comment:
What if, instead of using a constant here, you made the original `resolveKey` method return a nullable key. Then if the key is null, you don't update the timing, and you then don't need the AdvancedMonitorDecorator

Reply:
yeah thought about this also but this will mean to do that check all over the place in the Callback as there are many places there where this key should not be null, I would rather have this decorator than to use a nullable there that should not be null , what do you think ?

Reply:
I'm not a fan of overengineering things. I find it simpler to allow null rather than creating a decorator just for this behavior, but if you really think it's best, let's do it.